Web Feature Service part 2

nosy-b opened the pull request:
What this PR does:

* Load/parse a KML (a small sample can be found in exemples dir) 
* clamp lines and polygon to the ground
* generate icons and text boxes for 2D point 
* handle picking on polygons showing its attributes



_This Pull Request should have come after WFS integration #195  but that's a way to put pressure ;)_

gchoqueux commented:
@nosy-b  Warning, you have merged master (in ec776c51553dbf3c4ec32d6ca7f0ee0395038b71) in this branch kmlLoader. Prefer rebase

gchoqueux committed:
index ToolTip

gchoqueux committed:
fix (WMTS) : add default min/max zoom WMTS

gchoqueux committed:
refactor (Material) : rename material

gchoqueux committed:
Add size Vector2D in viewDiv

gchoqueux committed:
feature (material) : add RTC in lines and points
  * Points
  * Lines
  * PointMaterial
  * LineMaterial

gchoqueux committed:
review

gchoqueux committed:
refactor (content geometry) : content in NodeMesh
  * content is Map, where keys are layer.id

feat: Add getPickObject3d in c3Engine

gchoqueux committed:
feat (WFS) : new API functions

  * addFeatureLayer
  * addFeature
  * pickFeature
  * removeFeature

new WFS_Provider's features :
  * Point, box and line
  * Geojson parsing
  * 3d rge building

closes #115

gchoqueux committed:
Feat (WFS) : json config WFS
Create JSON objects for features and manage feature layers visibility

for of

gchoqueux committed:
GeoJson

gchoqueux committed:
added kml loading with clamp to ground
added picking in rasterized polygons
sample json added in example
changer nbSamplers min

gchoqueux committed:
rework buffering

gchoqueux commented:
moved in #450
